src/converters/helpers.js

```diff
@@ -1,5 +1,66 @@
+import * as path from 'node:path'
+import redirects from '@adguard/scriptlets/dist/redirects.json' with { type: 'json' }
+
+export function generateResourcesMapping() {
+  const resourcesMapping = new Map();
+  const allowedResourceExtensions = [
+    'html',
+    'js',
+    'css',
+    'mp4',
+    'mp3',
+    'xml',
+    'txt',
+    'json',
+    'empty',
+  ];
+
+  function getPreferredResource(aliases) {
+    for (let i = 0; i < aliases.length; i++) {
+      if (!allowedResourceExtensions.includes(aliases[i].split('.').pop())) {
+        continue;
+      }
+
+      // Skip manually created uBO aliases by AdGuard
+      if (aliases[i].startsWith('ubo-')) {
+        continue;
+      }
+
+      return aliases[i];
+    }
+
+    return null;
+  }
```

Comment:
I think that has the same semantic (only returning "undefined" instead of "null" (the calling checks for "null")

```suggestion
// ignore non-supported files and manually created uBO aliases by AdGuard
return aliases.find(alias => {
  const extension = alias.split('.').pop();
  return allowedResourceExtensions.includes(extension) && !alias.startsWith('ubo-');
});
```